pyqtgraph/graphicsItems/BoxplotItem.py

```diff
@@ -0,0 +1,304 @@
+        else:
+            loc = np.arange(len(data))
+        
+        locAsX = self.opts["locAsX"]
+        width = self.opts["width"]
+        pen = fn.mkPen(self.opts["pen"])
+        brush = fn.mkBrush(self.opts["brush"])
+        medianPen = fn.mkPen(self.opts["medianPen"])
+        
+        p = QtGui.QPainter(self.picture)
+        # for calculating bounding rect
+        pw = pen.widthF() * 0.7072
+        boxBounds = QRectF()
+        pixelPadding = 0
+        for pos, dataset in zip(loc, data):
+            dataset = np.asarray(dataset)
+            p75, median, p25 = np.percentile(dataset, [75, 50, 25])
+            lower, upper = self.whiskerFunc(dataset)
+            # get outlier data points if enabled
+            if self.opts["outlier"]:
+                mask = np.logical_or(dataset<lower, dataset>upper)
+                self.outlierData[pos] = dataset[mask]
+            
+            p.setPen(pen)
+            # whiskers
+            if locAsX:
+                p.drawLine(QPointF(pos-width/4, upper), QPointF(pos+width/4, upper))
+                p.drawLine(QPointF(pos-width/4, lower), QPointF(pos+width/4, lower))
+                p.drawLine(QPointF(pos, upper), QPointF(pos, p75))
+                p.drawLine(QPointF(pos, lower), QPointF(pos, p25))
+            else:
+                p.drawLine(QPointF(upper, pos-width/4), QPointF(upper, pos+width/4))
+                p.drawLine(QPointF(lower, pos-width/4), QPointF(lower, pos+width/4))
+                p.drawLine(QPointF(upper, pos), QPointF(p75, pos))
+                p.drawLine(QPointF(lower, pos), QPointF(p25, pos))
+            # box
+            p.setBrush(brush)
+            if locAsX:
+                p.drawRect(QRectF(pos-width/2, p25, width, p75-p25))
+            else:
+                p.drawRect(QRectF(p25, pos-width/2, p75-p25, width))
+            # median
+            p.setPen(medianPen)
+            if locAsX:
+                p.drawLine(QPointF(pos-width/2, median), QPointF(pos+width/2, median))
+            else:
+                p.drawLine(QPointF(median, pos-width/2), QPointF(median, pos+width/2))
+            # bounding rect
+            if locAsX:
+                rect = QRectF(pos-width/2, lower, width, upper-lower)
+            else:
+                rect = QRectF(lower, pos-width/2, upper-lower, width)
+            if pen.isCosmetic():
+                boxBounds |= rect
+                pixelPadding = max(pixelPadding, pw)
+            else:
+                boxBounds |= rect.adjusted(-pw, -pw, pw, pw)
+            
+        p.end()
+        self._boxBounds = boxBounds
+        self._pixelPadding = pixelPadding
+            
+    def paint(self, p, *args):
+        if self.picture is None:
+            self.generatePicture()
+        p.drawPicture(0, 0, self.picture)
+
+        if not self.opts["outlier"]:
+            return
+
+        # outlier related style
```

Comment:
one simplification to consider is to include a ScatterPlotItem as a child and delegate all the outlier drawing to it. But it's okay if you do not want to.